### PR TITLE
[Server] K8s connection management 

### PR DESCRIPTION
**Pull request by MUzairS15**
**Notes for Reviewers**

This PR fixes #
1. Show receipt when the kubeconfig is uploaded.
2. Manage contexts which have the state as `connected`. (Dynamic k8s registration).
   I have changed the logic to only query for context which has state as `connected`, this logic should handle all cases.
   The subsequent PRs will be handling the logic for controlling state transitions completely, i.e. in this PR state transitions are not supported. Operations like deploying operators will right now be controlled from UI, i.e. Backend does not have restrictions if UI requests an invalid transition.

In all the current functionality (as present in the master) is not affected but an additive change with improvements.
3. For UI, I have added an API to retrieve all possible state transitions for a particular connection kind.
**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by MUzairS15** [comment]
// @leecalcote @theBeginner86

**Comment by leecalcote** [comment]
I'm removing "[WIP]", @MUzairS15 ...

File: server/handlers/connections_handlers.go [hunk 1]
```diff
@@ -13,6 +13,8 @@ import (
 	"github.com/layer5io/meshkit/models/events"
 )
 
+type connectionStatusPayload map[uuid.UUID]models.ConnectionStatus
```
Comment (leecalcote) [inline server/handlers/connections_handlers.go:16]: @MUzairS15, does this map to the Status attribute in the Connections OpenAPI schema (are we being schema-driven)?

**Comment by leecalcote** [comment]
@MUzairS15, as the Connections page goes live in Meshery UI (@senali-d, @Yashsharma1911), the Connections API will need to respond with the available set of states to which each individual connection can transition given that connection's current state.

**Comment by leecalcote** [comment]
@MUzairS15 a few lint checks could use your attention.

**Comment by MUzairS15** [comment]
@yash37158 or @senali-d or @Akshay Sharma can you handle the styling here for the config upload modal?
@ritiksaxena124 Wil you ensure someones takes the above item?

**Comment by MUzairS15** [comment]
Screenshot 2023-11-01 at 8 20 40 PM

There are 3 lists, `Connected, Registered and Ignored contexts`, make them collapsible and make sure they follow the Meshery UI design.

**Comment by yash37158** [comment]
sure @MUzairS15, I will fix that

**Comment by MUzairS15** [comment]
Screenshot 2023-11-06 at 12 08 37 PM

// @leecalcote Merging .

**Comment by leecalcote** [comment]
@MUzairS15 where is the “?” that we discussed?

**Pull request merged**